src/channel.ts

```diff
@@ -787,6 +788,43 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
     }
   }
 
+  /**
+   * Sends an event to update the AI state for a specific message.
+   * Typically used by the server connected to the AI service to notify clients of state changes.
+   *
+   * @param messageId - The ID of the message associated with the AI state.
+   * @param state - The new state of the AI process (e.g., thinking, generating).
+   * @param options - Optional parameters, such as `ai_message`, to include additional details in the event.
+   */
+  async updateAIState(messageId: string, state: AIState, options: { ai_message?: string } = {}) {
+    await this.sendEvent({
+      ...options,
+      type: 'ai_indicator.update',
+      message_id: messageId,
+      ai_state: state,
+    } as Event<StreamChatGenerics>);
+  }
+
+  /**
+   * Sends an event to notify watchers to clear the typing/thinking UI when the AI response starts streaming.
+   * Typically used by the server connected to the AI service to inform clients that the AI response has started.
+   */
+  async clearAIIndicator() {
+    await this.sendEvent({
+      type: 'ai_indicator.clear',
```

Comment:
I am thinking, whether the low-level client should have opinion about what should happen in the UI.

Maybe instead of using word "clear" the name could indicate that streaming has started.

Reply:
I think the docs are wrong there, `ai.clear` comes after both `thinking` and `generating` have happened and concludes. At least that's how I implemented it (might be wrong). Will check in the actual node implementation in a bit

Reply:
Yes, clear is to clear the typing indicator shown in the UI. It basically means generation is completed.

Reply:
All questions are addressed in this Slack thread:
- https://getstream.slack.com/archives/CE5N802GP/p1732625926145299